File: daytwo/Day2.cs
using System;
using System.IO;

class Day2 {
    static void Main(string[] args) {
        partone();
        parttwo();
    }
    static void partone() {
        int x = 0;
        int depth = 0;
        foreach (string line in File.ReadAllLines("Day2.in")) {
            string[] tmp = line.Split(" ");
            if (tmp[0] == "forward") {
                x += Convert.ToInt32(tmp[1]);
            } else if (tmp[0] == "down") {
                depth += Convert.ToInt32(tmp[1]);
            } else {
                depth -= Convert.ToInt32(tmp[1]);
            }
        }
        Console.WriteLine(x*depth);
    }
    static void parttwo() {
        int aim = 0;
        int a = 0;
        int dep = 0;
        foreach (string line in File.ReadAllLines("Day2.in")) {
            string s = line.Split(" ")[0];
            Int32.TryParse(line.Split(" ")[1], out int x);
            if (s == "forward") {
                a += x;
                dep += aim * x;
            } else if (s == "down") {
                aim += x;
            } else {
                aim -= x;
            }
        }
        Console.WriteLine(a*dep);
    }
}

File: daytwo/Day2.cpp
#include "bits/stdc++.h"
using namespace std;
void partone(), parttwo();

int main() {
    partone();
    parttwo();
    return 0;
}

void partone() {
    ifstream fin = ifstream("Day2.in");
    ofstream fout = ofstream("Day2.out");
    int curdepth = 0;
    int curx = 0;
    string s;
    int x;
    while (fin >> s) {
        fin >> x;
        if (s == "forward") {
            curx += x;
        } else if (s == "down") {
            curdepth += x;
        } else {
            curdepth -= x;
        }
    }

    cout << curdepth * curx << '\n';

    fin.close();
    fout.close();
}

void parttwo() {
    ifstream fin = ifstream("Day2.in");
    ofstream fout = ofstream("Day2.out");
    int curdepth = 0;
    int curx = 0;
    int aim = 0;
    string s;
    int x;
    while (fin >> s) {
        fin >> x;
        if (s == "forward") {
            curx += x;
            curdepth += aim * x;
        } else if (s == "down") {
            aim += x;
        } else {
            aim -= x;
        }
    }


    cout << curdepth*curx << '\n';
    fin.close();
    fout.close();
}

/* stuff you should look for
    * int overflow, array bounds
    * special cases (n=1?)
    * do smth instead of nothing and stay organized
    * WRITE STUFF DOWN
    * CHECK IF DEBUGGING OUTPUT BLOCKS ANSWER
*/

